app/models/feedback.rb

```diff
@@ -1,3 +1,8 @@
 class Feedback < ApplicationRecord
   belongs_to :user
+
+  RATINGS = []
+  RATINGS << RATING_HAPPY = 'happy'
+  RATINGS << RATING_NEUTRAL = 'neutral'
```

Comment:
pourquoi on utilise pas une enum pour faire ça ?
Je crois qu'en effet kisskool rails t'engueule si tu essayes de mettre une valeur interdite

Reply:
Parce que les enums ça permet pas de facilement avoir une des valeurs de l'enum, alors qu'une constante oui. L'effet kiss-kool je fais pareil dans la prochaine PR avec un validateur (et en base une fois Rails 6 shippé)

Reply:
J'ai l'impression qu'on peux avoir la liste https://stackoverflow.com/questions/38904093/how-to-get-the-array-for-an-enum-model-attribute

Reply:
Yes mais justement, tu dois dire `MonEnum["clé"]`, donc si tu écris `MonEnum["clé-érronée"]`, pas de problème, ça va failer silently, alors que là avec constates ça gueule tout de suite.

Reply:
tu peux me donner un cas d'usage ? je n'ai pas compris ...

Reply:
En gros en passant par un enum, tu accès à une des valeurs de l'enum en faisant `MonModel.mon_enum["ma_clé"]`. Le problème avec ce fonctionnement c'est que si tu écris à la place de `"ma_clé"` une clé qui n'existe pas, le système ne plante pas, ça retourne simplement `nil`. Autrement dit, en cas de typo, on n'est pas protégé.

Alors qu'en utilisant des constantes, si tu écris le nom d'une constante qui n'existe pas, ça va planter donc tu vas bien être informé de ton erreur.

Reply:
Ex : 
```ruby
MonModel.mon_enum["ma_clé_inexistante"]
# => nil
MonModel::MA_CONSTANTE_INEXISTANTE
# => NameError (uninitialized constant MonModel::MA_CONSTANTE_INEXISTANTE)
```

Reply:
```
MonModel.mon_enum.fetch('ma_clé_inexistante')
# => KeyError
```
et en plus tu gagnes des scopes https://api.rubyonrails.org/classes/ActiveRecord/Enum.html

Reply:
En effet ! (je suis donc un peu moins fermé à convertir :) ).

Mais je vois quand même trois problèmes :
- c'est lourdingue `MonModel.mon_enum.fetch('ma_clé_inexistante')`
- il faut se contraindre à utiliser `fetch`, pas sur que tout le monde le fasse / l'attrape à la review
- Pour avoir toutes les valeurs (par exemple pour itérer dessus), faut faire `MonModel.mon_enum.keys` (pas hyper élégant je trouve)

Et enfin, je trouve ça un peu bizarre d'écrire un truc du genre
```ruby
enum rating: {
  happy:    'happy',
  neutral:  'neutral',
  unhappy:  'unhappy'
}
```
où on écrit 2 fois la même chose (clé = valeur)

Reply:
- c'est lourdingue : oui , mais en vrai, tu peux pas faire mauvais update sinon tu te fais taper par rails, et tu es sencé faire des where avec le scope buid in ... du coup, je sais pas quand tu as besoin du fetch ...
- fetch : voir plus haut
- toutes les valeurs `keys`, oui je trouve ca moche aussi
- l'écriture en double ... oui c'est chelou ... je me demande si on est vraiment obligé de l'utiliser

Finalement, t'as besoin d'une enum, rails fournit une enum : utilisons l'enum de rails !

Reply:
C'est bien parce que c'est toi... ;)